File: cart-service/src/main/java/com/fooddelivery/cartservice/model/CartItem.java
package com.fooddelivery.cartservice.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "cart_items")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cart_id", nullable = false)
    @JsonIgnore
    private Cart cart;

    @Column(nullable = false)
    private Long menuItemId;

    @Column(nullable = false)
    private String itemName; // Cache for display

    @Column(nullable = false)
    private Integer quantity;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price; // Price at time of adding to cart

    @Column(length = 500)
    private String customizations;
}

File: lib/models/cart_item.dart
class CartItem {
  final int id;
  final int menuItemId;
  final String itemName;
  final int quantity;
  final double price;
  final double itemTotal;
  final String? customizations;

  CartItem({
    required this.id,
    required this.menuItemId,
    required this.itemName,
    required this.quantity,
    required this.price,
    required this.itemTotal,
    this.customizations,
  });

  factory CartItem.fromJson(Map<String, dynamic> json) {
    return CartItem(
      id: json['id'],
      menuItemId: json['menuItemId'],
      itemName: json['itemName'],
      quantity: json['quantity'],
      price: (json['price'] as num).toDouble(),
      itemTotal: (json['itemTotal'] as num).toDouble(),
      customizations: json['customizations'],
    );
  }
}
